### DISCUSSION TITLE
[WidgetItem] GetIndex returns -1

### DISCUSSION THREAD

**Discussion by Icezman001** [runelite/runelite#15875] [Q&A] [answered]
I wasn't sure if i should make an actual issue for this or just on the Q&A so thought this would be a good starting point.

When I try to get the index of items in the inventory through widgetItem, all index's return -1 instead of the actual position in the inventory I would expect them to show.

On the API site it says:
`The index position of this WidgetItem inside its parents WidgetItem array.`

So I'm wondering if it's only because I'm using it wrong?
Does the inventory suddenly not have a WidgetItem array or is it accessed somewhere else?

Code to reproduce:
```
   @Override
    public void renderItemOverlay(Graphics2D graphics, int itemId, WidgetItem widgetItem) {
        if (plugin.getLocalPlayer() == null)
            return;

        widgetItem.getIndex();
        log.info("[" + widgetItem.getIndex() + "-" + itemId + "]");
    }
```

**Comment by Adam-** [comment, marked as answer]
getIndex() is specifically for if1 widgets, which the inventory isn't. Instead, use .getWidget().getIndex()

**Reply by Icezman001** [reply]
Thanks Adam! I had no idea it wasn't meant to be used with inventory
